### PR TITLE
Update agents.py

### PR DESCRIPTION
**Pull request by clavoirr**
these aren't really the bugs but some correct standard of writing python
*functions should be in lowercase and classes title in uppercase
*convert doc string in tripple double quoted strings doc
*replace assignment with argumented assignment
*remove redundant parentheses
*argument name should be lowercase

### CODE REVIEW

File: agents.py [hunk 1]
```diff
@@ -99,7 +98,7 @@ def can_grab(self, thing):
         return False
 
 
-def TraceAgent(agent):
+def trace_agent(agent):
```
Comment (norvig) [inline agents.py]: This is a factory method, so I think it is appropriate to be UpperCase.

File: agents.py [hunk 2]
```diff
@@ -108,13 +107,15 @@ def new_program(percept):
...
+
     agent.program = new_program
     return agent
 
+
 # ______________________________________________________________________________
 
 
-def TableDrivenAgentProgram(table):
+def table_driven_agent_program(table):
```
Comment (norvig) [inline agents.py]: Factory

File: agents.py [hunk 3]
```diff
@@ -125,55 +126,62 @@ def program(percept):
         percepts.append(percept)
         action = table.get(tuple(percepts))
         return action
+
     return program
 
 
-def RandomAgentProgram(actions):
-    "An agent that chooses an action at random, ignoring all percepts."
+def random_agent_program(actions):
```
Comment (norvig) [inline agents.py]: Factory

Comment (clavoirr) [inline agents.py]: surely they are but in factory methods also standard way is function name should start with lowercase

**Comment by clavoirr** [comment]
surely they are but in factory methods also standard way is function name should start with lowercase

**Comment by norvig** [comment]
Some good stuff here has been merged into #556.

**Pull request closed without merging**